src/raftstore/store/worker/region.rs

```diff
@@ -542,21 +611,121 @@ impl Runnable<Task> for Runner {
     }
 }
 
-impl RunnableWithTimer<Task, ()> for Runner {
-    fn on_timeout(&mut self, timer: &mut Timer<()>, _: ()) {
-        self.ctx.clean_timeout_ranges();
-        timer.add_task(Duration::from_millis(STALE_PEER_CHECK_INTERVAL), ());
+/// region related timeout event.
+pub enum Event {
+    CheckPeer,
+    CheckApply,
+}
+
+impl RunnableWithTimer<Task, Event> for Runner {
+    fn on_timeout(&mut self, timer: &mut Timer<Event>, event: Event) {
+        match event {
+            Event::CheckApply => {
+                if !self.pending_applies.is_empty() {
+                    if self.ctx.check_level_0_num_files() {
```

Comment:
A redundant check, `handle_pending_applies` can return a bool if it appled something.